kernel/src/platform/system_timer.rs

```diff
@@ -0,0 +1,139 @@
+//! Interface system tick timer.
+
+use crate::hil::time::{self, Frequency};
+
+/// Interface for the system tick timer.
+///
+/// A system tick timer provides a countdown timer to enforce process scheduling
+/// quantums.  Implementations should have consistent timing while the CPU is
+/// active, but need not operate during sleep.
+///
+/// On most chips, this will be implemented by the core (e.g. the ARM core systick), but
+/// some chips lack this optional peripheral, in which case it might be
+/// implemented by another timer or alarm controller, or require virtualization
+/// on top of a single hardware timer.
+pub trait SystemTimer {
+    /// Sets the timer as close as possible to the given interval in
+    /// microseconds, and starts counting down. Interrupts are not enabled.
+    ///
+    /// Callers can assume at least a 24-bit wide clock. Specific timing is
+    /// dependent on the driving clock. In practice, increments of 10ms are most
+    /// accurate and values up to 400ms are valid.
+    fn start_timer(&self, us: u32);
+
+    /// Returns if there is at least `us` microseconds left
+    fn greater_than(&self, us: u32) -> bool;
+
+    /// Returns true if the timer has expired since the last time this or set_timer()
+    /// was called. If called a second time without an intermittent call to set_timer(),
+    /// the return value is unspecified (implementations can return whatever they like)
+    fn expired(&self) -> bool;
+
+    /// Resets the timer
+    ///
+    /// Resets the timer to 0 and disables it
+    fn reset(&self);
+
+    /// Enable or disable interrupts
+    ///
+    fn config_interrupts(&self, enabled: bool);
+}
```

Comment:
```suggestion

    /// Return the number of microseconds remaining before the alarm expires.
    fn get_remaining_us(&self) -> u32;
}
```

You need that for the scheduler updates, right?

Reply:
yes, added.